contracts/VaultV3.vy

```diff
@@ -663,28 +649,31 @@ def _process_report(strategy: address) -> (uint256, uint256):
 
 @view
 @internal
-def _estimate_unlocked_profit() -> uint256:
+def _compute_unlocked_profit() -> uint256:
     """
-    If profit_distribution_rate is equal to zero, there is no profit to unlock, otherwise we estimate it
+    If profit_distribution_rate is equal to zero, there is no profit to unlock, otherwise we compute it
     """
-    if self.profit_distribution_rate != 0:
+    # Hack: to save gas
+    profit_distribution_rate: uint256 = self.profit_distribution_rate_
+    profit_last_update: uint256 = self.profit_last_update
+    if profit_distribution_rate != 0:
         if self.profit_end_date >= block.timestamp:
```

Comment:
this is an interesting scenario. you could cache `profit_end_date` for savings in the `else` (as you would be using twice). but taking into account the positive scenario should be a lot more common, probably not worth to cache it (as it would add 6 gas each time) 

probably worth adding a NOTE in comments

Reply:
added comment